server/src/controllers/aplus.ts

```diff
@@ -223,111 +223,112 @@ export const fetchAplusGrades = async (
       String(coursePartId)
     );
 
-    // TODO: There can be multiple sources
-    const gradeSource = await AplusGradeSource.findOne({
+    const gradeSources = await AplusGradeSource.findAll({
       where: {coursePartId: coursePart.id},
     });
```

Comment:
Add `as AplusGradeSourceData[]` and then you can skip some of the checks below because the cases can never happen